desc/equilibrium/coords.py

```diff
@@ -67,44 +71,66 @@ def map_coordinates(  # noqa: C901
 
     Returns
     -------
-    coords : ndarray, shape(k,3)
-        Coordinates mapped from inbasis to outbasis. Values of NaN will be returned
-        for coordinates where root finding did not succeed, possibly because the
-        coordinate is not in the plasma volume.
+    out : jnp.ndarray
+        Shape (k, 3).
+        Coordinates mapped from ``inbasis`` to ``outbasis``. Values of NaN will be
+        returned for coordinates where root finding did not succeed, possibly
+        because the coordinate is not in the plasma volume.
     info : tuple
         2 element tuple containing residuals and number of iterations
-        for each point. Only returned if ``full_output`` is True
-
-    Notes
-    -----
-    ``guess`` must be given for this function to be compatible with ``jit``.
+        for each point. Only returned if ``full_output`` is True.
 
     """
+    check_posint(maxiter, allow_none=False)
+    errorif(
+        not np.isfinite(tol) or tol <= 0,
+        ValueError,
+        f"tol must be a positive float, got {tol}",
+    )
+    params = setdefault(params, eq.params_dict)
     inbasis = tuple(inbasis)
     outbasis = tuple(outbasis)
-    assert (
-        np.isfinite(maxiter) and maxiter > 0
-    ), f"maxiter must be a positive integer, got {maxiter}"
-    assert np.isfinite(tol) and tol > 0, f"tol must be a positive float, got {tol}"
 
-    basis_derivs = tuple([f"{X}_{d}" for X in inbasis for d in ("r", "t", "z")])
+    if outbasis == ("rho", "theta", "zeta"):
+        # TODO: get iota if not supplied using below logic
+        if inbasis == ("rho", "alpha", "zeta") and "iota" in kwargs:
+            return _map_clebsch_coordinates(
+                coords,
+                kwargs.pop("iota"),
+                params["L_lmn"],
+                eq.L_basis,
+                guess,
```

Comment:
guess would be an nx3 array but i think we only want to pass in `guess[:,1]` here (and below)